paddle/contrib/inference/paddle_inference_api.h

```diff
@@ -12,49 +12,58 @@
    See the License for the specific language governing permissions and
    limitations under the License. */
 
+/*
+ * This file contains the definition of a simple Inference API for Paddle.
+ *
+ * ATTENTION: It requires some C++ features, for lower version C++ or C, we
+ * might release another API.
+ */
+
 #pragma once
 
+#include <memory>
 #include <string>
 #include <vector>
 
-namespace paddle {
+struct PaddleTensor {
+  std::string name;  // variable name.
+  std::vector<int> shape;
+  std::vector<unsigned char> data;         // bytes of data.
```

Comment:
this looks strange to me.
should it be 
```
std::unique_ptr<char> or perhaps just void* data, and user should delete it?
```

Reply:
The `data` is just a vector of bytes, the memory can be reused and resized, compared to `unique_ptr` memory.

Reply:
not sure user would want to to resize memory. I rarely see putting a blob of memory in vector. if not used correctly (e.g. push_back), the memory actually used can be unpredictable.

Reply:
The memory will be resized for different batch.